handlers/handler.go

```diff
@@ -0,0 +1,160 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/aws/aws-sdk-go/aws"
+	mq "github.com/remind101/mq-go"
+)
+
+type SQSHandler struct {
+	QueueURL   string
+	Start      bool
+	JobConfigs []JobConfig
+	Server     *mq.Server
+}
+
+type JobConfig struct {
+	Name        string      `name`
+	Pattern     string      `pattern`
+	Image       string      `image`
+	ImageConfig interface{} `image_config`
+}
+
+// NewSQSHandler creates new SQSHandler instance
+func NewSQSHandler(queueURL string) *SQSHandler {
+	sqsHandler := new(SQSHandler)
+	sqsHandler.QueueURL = queueURL
+	//sqsHandler.PatternMap = GetNewImagePatternMap()
+	return sqsHandler
+}
+
+// StartServer starts a server
+func (handler *SQSHandler) StartServer() error {
+	// return nil if the server already start
+	if handler.Server != nil {
+		return nil
+	}
+
+	fmt.Println("Start a new server")
+	newClient, err := NewSQSClient()
+	if err != nil {
+		return err
+	}
+
+	handler.Server = mq.NewServer(handler.QueueURL, mq.HandlerFunc(func(m *mq.Message) error {
+		return handler.HandleSQSMessage(m)
+	}), mq.WithClient(newClient))
+	handler.Server.Start()
+
+	return nil
+
+}
+
+// ShutdownServer shutdowns a server
+func (handler *SQSHandler) ShutdownServer() error {
+	fmt.Println("Shutdown the server")
+	if handler.Server == nil {
+		return nil
+	}
+	err := handler.Server.Shutdown(context.Background())
+	handler.Server = nil
+	return err
+}
+
+// HandleSQSMessage handles SQS message
```

Comment:
can we put docs about what the message body looks like and how does it handle case for error and retry?